procrustes/test/test_psdp.py

```diff
@@ -137,6 +175,18 @@ def test_psdp_peng_generic_non_square():
     assert_almost_equal(error, 32.21671819685297)
 
 
+def test_psdp_opt_generic_non_square():
+    r"""Test PSDP with 2 generic non-square matrices."""
+    a = np.array([[5, 1, 6, -1], [3, 2, 0, 2], [2, 4, 3, -3]])
+    b = np.array([[15, 1, 15 - 3, 2 + 5], [10, 5, 6, 3], [-3, 3, -3, -2 + 4]])
+    res = psdp_opt(a=a, b=b)
+    error = res["error"]
+    # Keeping error assertion to be less than a threshold value
+    # rather than "almost equal" to a given value because increasing
+    # the number of iterations might reduce the error obtained.
+    assert error <= 32.2
```

Comment:
Is there an actual positive semi-definite matrix provided?

Reply:
There is an issue when I provide that. Like the tests are often failing. Can we figure out the best way to resolve this on call?
@FanwangM

Reply:
Do you mean that there will be multiple positive-semi-definite matrix that makes the error less than 32.2? So, I would guess there is some randomness inside and we should provide a flag to control that.  @banrovegrie

Reply:
Yep, something like that. So when we use `assert_almost_equal()` wrt to the entire matrix it almost never passes.
@FanwangM